service/src/main/java/datawave/microservice/AccumuloConnectionService.java

```diff
@@ -0,0 +1,220 @@
+        connection.setAuths(getAuths(user));
+        connection.setAccumuloClient(accumuloClient);
+        return connection;
+    }
+    
+    private String getSupplierValueIfBlank(final String value, final Supplier<String> supplier) {
+        return StringUtils.isBlank(value) ? supplier.get() : value;
+    }
+    
+    /**
+     * Return the accumulo authorizations for the user
+     *
+     * @param currentUser
+     *            the user for whom to get the authorizations
+     *            
+     * @return the accumulo authorizations for the user
+     */
+    public Set<Authorizations> getAuths(DatawaveUserDetails currentUser) {
+        //@formatter:off
+        return currentUser.getProxiedUsers().stream()
+                .map(DatawaveUser::getAuths)
+                .map(a -> new Authorizations(a.toArray(new String[0])))
+                .collect(Collectors.toSet());
+        //@formatter:on
+    }
+    
+    /**
+     * Return the authorizations for the specified user downgraded to the specified level
+     *
+     * @param requestedAuthorizations
+     *            the level of authorizations to downgrade to
+     * @param currentUser
+     *            the user for whom to downgrade the authorizations
+     *            
+     * @return the authorizations for the specified user downgraded to the specified level
+     */
+    public Set<Authorizations> getDowngradedAuthorizations(String requestedAuthorizations, DatawaveUserDetails currentUser) {
+        DatawaveUser primaryUser = currentUser.getPrimaryUser();
+        return userAuthFunctions.mergeAuthorizations(userAuthFunctions.getRequestedAuthorizations(requestedAuthorizations, currentUser.getPrimaryUser()),
+                        currentUser.getProxiedUsers(), u -> u != primaryUser);
+    }
+    
+    /**
+     * Return the keys. If a regexTerm is specified, the keys will be filtered according to that term, otherwise all the keys in the specified table that the
+     * user has authorizations for will be returned.
+     *
+     * @param modelTable
+     *            the model table to get the keys of
+     * @param currentUser
+     *            the user requesting the keys. Used for authorizations purposes
+     * @param regexTerm
+     *            the regex (can be just a string) to apply to the IteratorSetting
+     *            
+     * @return the keys of hte specified table that the specified user has the authorzations to access.
+     * @throws TableNotFoundException
+     *             Thrown if the table is not found
+     */
+    public List<Key> getKeys(String modelTable, DatawaveUserDetails currentUser, String regexTerm) throws TableNotFoundException {
+        Scanner scanner = ScannerHelper.createScanner(this.accumuloClient, modelTable, this.getAuths(currentUser));
+        if (!regexTerm.isEmpty()) {
+            IteratorSetting cfg = new IteratorSetting(21, "colfRegex", RegExFilter.class.getName());
```

Comment:
I wonder how many of the calls to this require a regex and if they can be replaced with an exact colf value which could be more performance depending on locality groups.

Reply:
I think this is representative of the original logic here: https://github.com/NationalSecurityAgency/datawave/blob/integration/web-services/model/src/main/java/datawave/webservice/query/model/ModelBean.java#L356

This was part of some refactor work Ceria did to move the model rest endpoint to the dictionary service.